### ISSUE TITLE
Frequent STP failures on Azure Pipelines

### ISSUE THREAD

**Issue by gsnedders** [web-platform-tests/wpt#27381]
[Meta triage issue]

I was wondering why it took so long to get STP119 results, and it appears we have a bunch of flakiness.

Looking at [these failures](https://dev.azure.com/web-platform-tests/wpt/_build?definitionId=1&_a=summary&branchFilter=1091%2C9077%2C9077&statusFilter=warning%2Cfailed%2Ccancelled) currently we have 66 failures dating back to 31 Dec 2020. Note that not all of these are Safari (some are Edge, some are hooks related to WPT.fyi upload, and I've not counted rare failures clearly unrelated to Safari—like Python package installation)

Of these, there are four likely Azure-related causes:

 * Agent stopped responding: 4 jobs / 3 runs
 * "Received request to deprovision: The request was cancelled by the remote provider": 1 jobs / 1 runs
 * "An error occurred while provisioning resources (Error Type: Disconnect).": 4 jobs / 4 runs
 * "The operation was canceled.": 1 jobs / 1 runs (e.g. [this run](https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=59990&view=logs&j=9e909769-fc48-58b9-7383-225ac465e77e&t=915084e6-0140-5839-bf69-6179669ce085))

And three likely Safari-related causes:

 * Unable to locate safaridriver binary: 93 jobs / 23 runs (e.g. [this](https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=59186&view=logs&j=42513d51-f539-5b97-080a-60c327907e2e) & [that](https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=59773&view=logs&j=d41515ed-d71e-5e8d-de88-940c2a934d56&t=90db94aa-13d6-56f8-7ca7-ab92a82fe450))
 * Exceeded timeout: 12 jobs / 11 runs (split pretty evenly across the different chunks)
 * Max restarts exceeded: 2 jobs / 2 runs ([this](https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=60066&view=logs&j=42513d51-f539-5b97-080a-60c327907e2e&t=36740ece-039c-5053-573a-2f567ef39915) & [that](https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=60159&view=logs&j=fc030ca2-9b33-54ac-940f-f7bd4c891f5c&t=cd1c211f-fc03-57c9-60bb-e01678c77ad3&l=195256))

Further discussion to follow in comments below.

**Comment by gsnedders** [comment]
> Unable to locate safaridriver binary: 93 jobs / 23 runs (e.g. this & that)

Most of this has now been fixed with #27226, but there were occasional failures—mostly longer ago—where it followed something like:

```
2021-01-06T01:35:02.5366850Z ##[section]Starting: Install Safari Technology Preview
2021-01-06T01:35:02.5375110Z ==============================================================================
2021-01-06T01:35:02.5375680Z Task         : Command line
2021-01-06T01:35:02.5376210Z Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
2021-01-06T01:35:02.5376670Z Version      : 2.178.0
2021-01-06T01:35:02.5377040Z Author       : Microsoft Corporation
2021-01-06T01:35:02.5377550Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
2021-01-06T01:35:02.5378100Z ==============================================================================
2021-01-06T01:35:02.7905820Z Generating script.
2021-01-06T01:35:02.7911000Z ========================== Starting Command Output ===========================
2021-01-06T01:35:02.8317260Z [command]/bin/bash --noprofile --norc /Users/runner/work/_temp/b71cc10a-284e-4a3f-b701-993d60f7bcbc.sh
2021-01-06T01:35:10.9276700Z Warning: Calling brew cask install is deprecated! Use brew install [--cask] instead.
2021-01-06T01:35:11.2385430Z ==> Downloading https://secure-appldnld.apple.com/STP/001-86197-20201209-2fd0fe89-0c37-412d-99e2-4e288519e886/SafariTechnologyPreview.dmg
2021-01-06T01:35:11.8569110Z curl: (56) LibreSSL SSL_read: SSL_ERROR_SYSCALL, errno 54
2021-01-06T01:35:11.8807760Z Error: Download failed on Cask 'safari-technology-preview' with message: Download failed: https://secure-appldnld.apple.com/STP/001-86197-20201209-2fd0fe89-0c37-412d-99e2-4e288519e886/SafariTechnologyPreview.dmg
2021-01-06T01:35:12.9333110Z ##[section]Finishing: Install Safari Technology Preview
```

In both that case and #27226 we should really be failing the job before we even get to the running task, the install task itself should fail.

**Comment by gsnedders** [comment]
> Exceeded timeout: 12 jobs / 11 runs (split pretty evenly across the different chunks)
> Max restarts exceeded: 2 jobs / 2 runs (this & that)

The latter at least is likely to be solved by #19620; the former is harder to tell given on the face of it it is just things getting very slow (why?).

The latter ends with multiple restarts failing like:

```
2021-01-26T17:14:08.5351160Z 81:32.01 INFO Starting runner
2021-01-26T17:14:30.8964640Z 81:54.36 WARNING Failed to start protocol connection
2021-01-26T17:14:30.9248540Z 81:54.36 WARNING Traceback (most recent call last):
2021-01-26T17:14:30.9350890Z   File "/Users/runner/work/1/s/tools/wptrunner/wptrunner/executors/protocol.py", line 47, in setup
2021-01-26T17:14:30.9714140Z     self.connect()
2021-01-26T17:14:30.9821470Z   File "/Users/runner/work/1/s/tools/wptrunner/wptrunner/executors/executorwebdriver.py", line 322, in connect
2021-01-26T17:14:30.9823470Z     self.webdriver.start()
2021-01-26T17:14:30.9926870Z   File "/Users/runner/work/1/s/tools/webdriver/webdriver/client.py", line 527, in start
2021-01-26T17:14:31.0027740Z     value = self.send_command("POST", "session", body=body)
2021-01-26T17:14:31.0133680Z   File "/Users/runner/work/1/s/tools/webdriver/webdriver/client.py", line 568, in send_command
2021-01-26T17:14:31.0229430Z     response = self.transport.send(
2021-01-26T17:14:31.0355560Z   File "/Users/runner/work/1/s/tools/webdriver/webdriver/transport.py", line 238, in send
2021-01-26T17:14:31.0456040Z     response = self._request(method, uri, payload, headers, timeout=None)
2021-01-26T17:14:31.0557970Z   File "/Users/runner/work/1/s/tools/webdriver/webdriver/transport.py", line 263, in _request
2021-01-26T17:14:31.0775230Z     response = self.connection.getresponse()
2021-01-26T17:14:31.1185650Z   File "/Users/runner/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/http/client.py", line 1347, in getresponse
2021-01-26T17:14:31.1188950Z     response.begin()
2021-01-26T17:14:31.1843110Z   File "/Users/runner/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/http/client.py", line 307, in begin
2021-01-26T17:14:31.1844000Z     version, status, reason = self._read_status()
2021-01-26T17:14:31.1844540Z   File "/Users/runner/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/http/client.py", line 276, in _read_status
2021-01-26T17:14:31.1845320Z     raise RemoteDisconnected("Remote end closed connection without"
2021-01-26T17:14:31.1845930Z http.client.RemoteDisconnected: Remote end closed connection without response
2021-01-26T17:14:31.1846280Z 
2021-01-26T17:14:31.2365090Z 81:54.36 WARNING Command left in command_queue during cleanup: 'init_failed', ()
2021-01-26T17:14:31.2366570Z 81:54.36 INFO Closing logging queue
2021-01-26T17:14:31.2367360Z 81:54.36 INFO queue closed
```

To me the most interesting/bizarre part of this is that we're seemingly never reading the init_failed message from the queue, or are the two timeouts just racing?
